dissect/target/plugins/os/windows/dpapi/crypto.py

```diff
@@ -140,158 +114,135 @@ def __init_subclass__(cls):
 
     This implementation relies on a faulty implementation from Microsoft that does not respect the HMAC RFC.
-    Instead of updating the inner pad, we update the outer pad...
-    This algorithm is also used when checking the HMAC for integrity after decryption
+    Instead of updating the inner pad, we update the outer pad.
+    This algorithm is also used when checking the HMAC for integrity after decryption.
+
     Args:
-        masterkey: decrypted masterkey (should be 64 bytes long)
-        nonce: this is the nonce contained in the blob or the HMAC in the blob (integrity check)
-        hash_algo: a HashAlgorithm to use for calculating block sizes
-        entropy: this is the optional entropy from CryptProtectData() API
-        strong_password: optional password used for decryption or the blob itself
-        smart_card_secret: optional MS Next Gen Crypto secret (e.g. from PIN code)
-        verif_blob: optional encrypted blob used for integrity check
+        master_key: Decrypted master key (should be 64 bytes long).
+        nonce: This is the nonce contained in the blob or the HMAC in the blob (integrity check).
+        hash_algorithm: A :class:`HashAlgorithm` to use for calculating block sizes.
+        entropy: This is the optional entropy from ``CryptProtectData()`` API.
+        strong_password: Optional password used for decryption or the blob itself.
+        smart_card_secret: Optional MS Next Gen Crypto secret (e.g. from PIN code).
+        verify_blob: Optional encrypted blob used for integrity check.
     Returns:
         decryption key
     """
-    if len(masterkey) > 20:
-        masterkey = hashlib.sha1(masterkey).digest()
+    if len(master_key) > 20:
+        master_key = hashlib.sha1(master_key).digest()
 
-    masterkey += b"\x00" * hash_algo.block_length
-    pad1 = "".join(chr(masterkey[i] ^ 0x36) for i in range(hash_algo.block_length))
-    pad2 = "".join(chr(masterkey[i] ^ 0x5C) for i in range(hash_algo.block_length))
+    master_key = master_key.ljust(hash_algorithm.block_length, b"\x00")
+    pad1 = bytes(c ^ 0x36 for c in master_key)
+    pad2 = bytes(c ^ 0x5C for c in master_key)
 
-    digest1 = hashlib.new(hash_algo.name)
-    digest1.update(pad2.encode("latin1"))
+    digest1 = hashlib.new(hash_algorithm.name)
+    digest1.update(pad2)
 
-    digest2 = hashlib.new(hash_algo.name)
-    digest2.update(pad1.encode("latin1"))
+    digest2 = hashlib.new(hash_algorithm.name)
+    digest2.update(pad1)
     digest2.update(nonce)
-    if smart_card_secret is not None:
+
+    if entropy and smart_card_secret:
         digest2.update(entropy + smart_card_secret)
-        if verif_blob is not None:
-            digest2.update(verif_blob)
+        if verify_blob:
+            digest2.update(verify_blob)
 
     digest1.update(digest2.digest())
-    if entropy is not None and smart_card_secret is None:
+    if entropy and smart_card_secret:
```

Comment:
There is a difference between:
```python
if entropy is not None and smart_card_secret is None:
# entropy = b"a"
# smart_card_secret = None
# If passes
```
and 
```python
if entropy and smart_card_secret:
# entropy = b"a"
# smart_card_secret = None
# If does not pass
```
I believe this should be:
```python
if entropy and not smart_card_secret:
```

Reply:
Ah I misread this as `smart_card_secret is not None` I think!